R/install.R

```diff
@@ -582,6 +582,16 @@ renv_install_package_impl <- function(record, quiet = TRUE) {
   else
     r_cmd_install(package, path)
 
+  # if we just installed a binary package, check that it can be loaded
+  # (source packages are checked by default on install)
+  if (isbin) {
+    status <- catch(renv_install_test(package))
```

Comment:
Perhaps we should only do this step on Linux? Or is there some fear that binaries from Windows or macOS could fail to load in some scenarios?

Reply:
(This might theoretically be relevant depending on how binaries get built on PPM?)

Reply:
It is possible to "accidentally" use the wrong binary package in any environment.

From https://bugs.r-project.org/show_bug.cgi?id=18549:

```r
lib <- file.path(getwd(), "lib")
dir.create(lib)
.libPaths(c(lib, .libPaths()))
install.packages("https://cran.r-project.org/bin/macosx/big-sur-arm64/contrib/4.2/Rcpp_1.0.10.tgz")
# installation is permitted without error
library(Rcpp)
#> Error: package or namespace load failed for ‘Rcpp’ in dyn.load(file, DLLpath = DLLpath, ...):
#>  unable to load shared object '/Users/aron/rcpp/lib/Rcpp/libs/Rcpp.so':
#>   dlopen(/Users/aron/rcpp/lib/Rcpp/libs/Rcpp.so, 0x0006): tried: '/Users/aron/rcpp/lib/Rcpp/libs/Rcpp.so' (mach-o file, but is an incompatible architecture (have (arm64), need (x86_64h)))
#> In addition: Warning message:
#> package ‘Rcpp’ was built under R version 4.2.0
```

That said, my experience is mostly with Linux packages and these problems occurring because folks cross-wire distribution binaries or lack system dependencies.

Reply:
What about adding an option to let folks disable this step, then we can control the per-OS default and let folks override that default choice if they choose.

Reply:
Right now you can opt out with `options(INSTALL_opts = "--no-test-load")`, which is what you would normally do for source package installation as well.

Reply:
That's fair. I was thinking that the load-test of binary packages might want to be controlled by a separate toggle, since it's a new feature and different than the default behavior in R.

I was also thinking about the Packrat side, where we will want a global toggle, just to give us an "out" in case something unexpected appears in some external installation.

Reply:
That's a good idea. I like escape hatches.